src/VisualStudio/Core/Def/Diagnostics/VisualStudioCodeAnalysisSuggestionsConfigService.cs

```diff
@@ -0,0 +1,231 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeAnalysisSuggestions;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.CodeAnalysisSuggestions;
+
+[ExportWorkspaceService(typeof(ICodeAnalysisSuggestionsConfigService), ServiceLayer.Host), Shared]
+[Export(typeof(VisualStudioCodeAnalysisSuggestionsConfigService))]
+internal sealed partial class VisualStudioCodeAnalysisSuggestionsConfigService : ICodeAnalysisSuggestionsConfigService, IDisposable
+{
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public VisualStudioCodeAnalysisSuggestionsConfigService(
+        VisualStudioWorkspace workspace,
+        IGlobalOptionService globalOptions,
+        IDiagnosticAnalyzerService diagnosticAnalyzerService)
+    {
+        _workspace = workspace;
+        _globalOptions = globalOptions;
+        _diagnosticAnalyzerService = diagnosticAnalyzerService;
+        _documentTrackingService = workspace.Services.GetRequiredService<IDocumentTrackingService>();
+
+        workspace.WorkspaceChanged += Workspace_WorkspaceChanged;
+    }
+
+    private readonly VisualStudioWorkspace _workspace;
+    private readonly IGlobalOptionService _globalOptions;
+    private readonly IDocumentTrackingService _documentTrackingService;
+    private readonly IDiagnosticAnalyzerService _diagnosticAnalyzerService;
+
+    public Task InitializeAsync(CancellationToken cancellationToken)
+    {
+        RefreshDiagnostics(_workspace.CurrentSolution, cancellationToken);
+        return Task.CompletedTask;
+    }
+
+    private void RefreshDiagnostics(Solution solution, CancellationToken cancellationToken)
+    {
+        foreach (var project in solution.Projects)
+            KickOffBackgroundComputation(project, cancellationToken);
+    }
+
+    private void KickOffBackgroundComputation(Project project, CancellationToken cancellationToken)
+        => Task.Run(() => GetCodeAnalysisSuggestionsConfigDataAsync(project, forceCompute: true, cancellationToken), cancellationToken);
```

Comment:
this reallllly is problematic for me.  this feels like we're entirely moving away from a pull model, and going back to automatic computation of expensive work.